### PR TITLE
Feature/ KNITE-1252 UI changes

### PR DESCRIPTION
**Pull request by abin-aloysious**
Contains fixes for the tickets:
1. KNITE-1252: UI changes in custom-styles hierarchy
2. KNITE-1240: Style Editor "Normal" style should not have Hierarchy available
3. KNITE-1292: 'strong' tag issue in custom styles
4. KNITE-1310: After creating an "prepend" style in Style Editor, the style cannot be altered

### CODE REVIEW

File: src/ui/CustomStyleEditor.tsx [hunk 1]
```diff
@@ -274,7 +275,7 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
               this.state.styles.prefixValue
             )}</strong>${textSample}`;
           } else {
-            sampleDiv.innerText = `${this.getNumberingLevel(
+            sampleDiv.innerHTML = `${this.getNumberingLevel(
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:278]: Why this change? The value assigned doesn't look like html, and if it does need to be html, it should use dom elements instead of innerHTML to avoid injection attacks.

File: src/ui/CustomStyleEditor.tsx [hunk 2]
```diff
@@ -456,10 +457,14 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
   // [FS] IRAD-1127 2020-12-31
   // to populate the selected custom styles.
   onSelectCustomStyle(e) {
-    if (null !== customStyles) {
+    if (customStyles !== null) {
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:460]: This should just be a truthy check. The next line will hard fail if customStyles is `undefined`

File: src/ui/CustomStyleEditor.tsx [hunk 3]
```diff
@@ -456,10 +457,14 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
...
       const value = customStyles.find((u) => u.styleName === e.target.value);
+
       // FIX: not able to modify and save the populated style
-      value?.mode !== undefined && (value.mode = 3); //FSFIX
+      if (value?.mode !== undefined) {
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:464]: can be simplified to nullish assignment `value.mode ??= 3`

File: src/ui/CustomStyleEditor.tsx [hunk 4]
```diff
@@ -540,23 +545,21 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
...
   handleList(val) {
-    //edit mode
-    if (
-      this.state.mode > 0 &&
-      this.isCustomStyleAlreadyApplied() &&
-      false === val.target.checked
-    ) {
+    const selectedStyle = val.target.value;
+    const isList = selectedStyle === 'listStyle';
+    const styleLevel = selectedStyle === 'none' ? 0 : isList ? 1 : 1;
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:550]: ` isList ? 1 : 1` will always return `1`. confusing and probably a bug.

File: src/ui/CustomStyleEditor.tsx [hunk 5]
```diff
@@ -540,23 +545,21 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
...
+    if (this.state.mode > 0 && this.isCustomStyleAlreadyApplied() && !isList) {
       this.showAlert();
     } else {
       this.setState((prevState) => ({
+        selectedStyle,
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:555]: shouldn't this start with `...prevState` to make sure any other state values not overwritten are preserved?

File: src/ui/CustomStyleEditor.tsx [hunk 6]
```diff
@@ -639,16 +642,28 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
         }
       });
     }
+
+    if (this.state.styleName === RESERVED_STYLE_NONE) {
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:646]: Can this style change not be handled with css?

File: src/ui/CustomStyleEditor.tsx [hunk 7]
```diff
@@ -1212,33 +1227,33 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
...
-                    >
+                    <div className="molsp-hierarchydiv">
+                      <label>
+                        <input
+                          checked={!this.state.styles.isList}
+                          disabled={
+                            this.state.disableControl ||
+                            this.state.styleName === RESERVED_STYLE_NONE
+                          }
+                          onChange={this.handleList.bind(this)}
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:1238]: prefer arrow function over bind for closure. (echo for all binds)

File: src/ui/CustomStyleEditor.tsx [hunk 8]
```diff
@@ -1465,11 +1575,13 @@ export class CustomStyleEditor extends React.PureComponent<any, any> {
                   </div>
                   <div className="molsp-settingsdiv">
                     <input
+                      disabled={
+                        this.state.styles.isList === true
```
Comment (Tezra) [inline src/ui/CustomStyleEditor.tsx:1579]: ` === true` is redundent

**Review by Tezra — CHANGES_REQUESTED** [review]

File: src/ui/CustomStyleEditor.tsx (hunk 1 above)
Comment (abin-aloysious) [reply src/ui/CustomStyleEditor.tsx:278]: This update addresses ticket KNITE-1292: Fix for <strong> tag issue in custom styles.

Previously, when the bold style was applied to the first word or sentence, the <strong> tag would be added incorrectly at the word or sentence level instead of applying the bold formatting. This issue has now been resolved with this fix.

Comment (Tezra) [reply src/ui/CustomStyleEditor.tsx:278]: @abin-aloysious as far as I can tell though innerText will suffice here. If you must add html, use document.createElement + append. You can only use innerHTML with static hardcoded strings for security reasons.

**Comment by Tezra** [comment]
@abin-aloysious Lint isn't running and needs resync. Also, cannot approve PR until the "innerHTML" vulnerability is fixed/reverted.

**Review by Tezra — CHANGES_REQUESTED** [review]
@abin-aloysious Lint isn't running and needs resync. Also, cannot approve PR until the "innerHTML" vulnerability is fixed/reverted.

**Review by Tezra — APPROVED** [review]
